From f34521b0b1a0e3a638b55015c3f5ccfcd22e6dfb Mon Sep 17 00:00:00 2001
From: Jonathan M Hsieh <jmhsieh@apache.org>
Date: Mon, 2 Nov 2015 05:03:38 -0800
Subject: [PATCH 273/314] HBASE-14731 Add -DuseMob option to ITBLL

Reason: Testing
Author: Jonathan M Hsieh
Ref: CDH-26063

Conflicts:
	hbase-it/src/test/java/org/apache/hadoop/hbase/test/IntegrationTestBigLinkedList.java

Change-Id: Ib067de4da8aa4f7fd587f162c11572d4c6e4debe
(cherry picked from commit 80b676fd68f63c76812d81900d7439aec9013a1b)
---
 .../hbase/test/IntegrationTestBigLinkedList.java   |   21 ++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/hbase-it/src/test/java/org/apache/hadoop/hbase/test/IntegrationTestBigLinkedList.java b/hbase-it/src/test/java/org/apache/hadoop/hbase/test/IntegrationTestBigLinkedList.java
index e43881a..4450d6d 100644
--- a/hbase-it/src/test/java/org/apache/hadoop/hbase/test/IntegrationTestBigLinkedList.java
+++ b/hbase-it/src/test/java/org/apache/hadoop/hbase/test/IntegrationTestBigLinkedList.java
@@ -484,6 +484,14 @@ public class IntegrationTestBigLinkedList extends IntegrationTestBase {
         if (!admin.tableExists(tableName)) {
           HTableDescriptor htd = new HTableDescriptor(getTableName(getConf()));
           htd.addFamily(new HColumnDescriptor(FAMILY_NAME));
+          // if -DuseMob=true force all data through mob path.
+          if (conf.getBoolean("useMob", false)) {
+            for (HColumnDescriptor hcd : htd.getColumnFamilies() ) {
+              hcd.setMobEnabled(true);
+              hcd.setMobThreshold(4);
+            }
+          }
+
           int numberOfServers = admin.getClusterStatus().getServers().size();
           if (numberOfServers == 0) {
             throw new IllegalStateException("No live regionservers");
@@ -1446,6 +1454,19 @@ public class IntegrationTestBigLinkedList extends IntegrationTestBase {
     System.err.println(" loop       Program to Loop through Generator and Verify steps");
     System.err.println(" clean      Program to clean all left over detritus.");
     System.err.println(" search     Search for missing keys.");
+    System.err.println("");
+    System.err.println("General options:");
+    System.err.println(" -D"+ TABLE_NAME_KEY+ "=<tableName>");
+    System.err.println("    Run using the <tableName> as the tablename.  Defaults to "
+        + DEFAULT_TABLE_NAME);
+    System.err.println(" -D"+ HBaseTestingUtility.REGIONS_PER_SERVER_KEY+ "=<# regions>");
+    System.err.println("    Create table with presplit regions per server.  Defaults to "
+        + HBaseTestingUtility.DEFAULT_REGIONS_PER_SERVER);
+
+    System.err.println(" -DuseMob=<true|false>");
+    System.err.println("    Create table so that the mob read/write path is forced.  " +
+        "Defaults to false");
+
     System.err.flush();
   }
 
-- 
1.7.9.5

